onadata/apps/logger/models/instance.py

```diff
@@ -141,6 +142,35 @@ def update_user_submissions_counter(instance, created, **kwargs):
     queryset.update(count=F('count') + 1)
 
 
+def update_xforms_monthly_submission_counter(instance, created, **kwargs):
+    if not created:
+        return
+    if getattr(instance, 'defer_counting', False):
+        return
+
+    user_id = XForm.objects.values_list('user_id', flat=True).get(
+        pk=instance.xform_id
+    )
+    date_created = instance.date_created
+    first_day_of_month = date(
+        year=date_created.year,
+        month=date_created.month,
+        day=1
+    )
+    queryset = XFormSubmissionCounter.objects.filter(
+        user_id=user_id,
+        xform=instance.xform,
+        timestamp=first_day_of_month,
+    )
+    if not queryset.exists():
+        XFormSubmissionCounter.objects.create(
+            user_id=user_id,
+            xform=instance.xform,
+            timestamp=first_day_of_month,
+        )
```

Comment:
there's a race condition here between `exists()` and `create()`. one solution would be to try `create()` unconditionally, and if it fails because it would violated a unique constraint, then `pass`

Reply:
the approach of `get_or_create()` followed by `update(F())` used in https://github.com/kobotoolbox/kobocat/pull/806/ is also fine